pkg/controller/webhooks/postgresql.go

```diff
@@ -287,3 +298,146 @@ func validateVacuumRepack(vacuum, repack bool) error {
 	}
 	return nil
 }
+
+func validatePgConf(pg *vshnv1.VSHNPostgreSQL) (fErros field.ErrorList) {
+
+	pgConfBytes := pg.Spec.Parameters.Service.PostgreSQLSettings
+
+	pgConf := map[string]string{}
+	if pgConfBytes.Raw != nil {
+		err := json.Unmarshal(pgConfBytes.Raw, &pgConf)
+		if err != nil {
+			fErros = append(fErros, &field.Error{
+				Field:    "spec.parameters.service.postgresqlSettings",
+				Detail:   fmt.Sprintf("Error parsing pgConf: %s", err.Error()),
+				Type:     field.ErrorTypeInvalid,
+				BadValue: pgConfBytes,
+			})
+			return fErros
+		}
+	}
+
+	for key, _ := range pgConf {
+		if key == "listen_addresses" {
```

Comment:
Can you please define a list of forbidden pg parameters on top of this file and instead of having multiple if clauses to just have one? I would also consider create a new go file to separate the concerns and make it a bit more readable

Reply:
I agree, having a map of forbidden keys would be more readable. You can use the maps key for the field name, so you can efficiently do checks. Then use the value of the map to store the specifics of the error.

It would also be easier to extend or change in the future.

Reply:
The error seems to be almost the same, only the name is added to a field. I would still go with just a list.